### ISSUE TITLE
ES6 Support

### ISSUE THREAD

**Issue by jadengore** [chriszarate/bookmarkleter#7]
Since `uglify` depends on ES5 code, the web utility makes it impossible for me to create a bookmarklet from ES6 code.

@chriszarate I'm curious about how this would go about being patched. I have a couple ideas:
1. Continue using Uglify, and run the code to be transformed through a compiler like Babel in the bookmarkleter library.
2. Optionally allow code to be parsed by Babel in the browser tool so that ES6 can work.
3. Pick a different minifier, such as [Babili](https://babeljs.io/blog/2016/08/30/babili), to process minification. This will allow ES6 to work out of the box.

I'd be willing to implement this, but I wanted to get your feedback before I do any serious work.

**Comment by chriszarate** [comment]
Thanks for raising this ... I hadn't really thought about it.

Bundling a ES*->ES5 compiler as part of the lib or browser tool strikes me as something we might regret considering how vast and modular that ecosystem is. We'd inevitably get complaints that this or that plugin or preset isn't supported. I lean towards pointing them elsewhere for compiling.

That said, providing the option of using Babili, in both the lib and browser-tool, sounds great. In the browser, perhaps we could even auto-detect ES6+ code and alert them to that option?

**Comment by alanhogan** [comment]
+1. I wanted my bookmarklet to work in IE 11 (don’t ask), so I babel-repl'd my code before pasting; it would be nice to have a more convenient option

**Comment by jadengore** [comment]
@alanhogan my fork has basic ES6 support, see here: http://jadengore.github.io/bookmarkleter

**Comment by chriszarate** [comment]
@jadengore PR?

**Comment by jadengore** [comment]
@chriszarate I wasn't perfectly happy with the solution I came up with, but I would appreciate your feedback. I'll open it now.

**Comment by jadengore** [comment]
This was resolved by #19, closing.
